docs/ops/matrix/MatMul_1.md

```diff
@@ -12,29 +12,43 @@
+1. Applying transpositions specified by optional `transpose_a` and `transpose_b` attributes for 2D tensors and bigger.
 
-2. Applied transpositions specified by optional `transpose_a` and `transpose_b` attributes.
+2. One-dimensional tensors unsqueezing is applied for each input independently. The axes inserted in this step are not included in the output shape.
+    * If rank of the **first** input is less than 2, it is unsqueezed to 2D tensor by adding axes with size 1 at ROW_INDEX_DIM, to the **left** of the shape. For example `[S]` will be reshaped to `[1, S]`.
+    * If rank of the **second** input is less than 2, it is unsqueezed to 2D tensor by adding axes with size 1 at COL_INDEX_DIM, to the **right** of the shape. For example `[S]` will be reshaped to `[S, 1]`.
 
 3. If ranks of input arguments are different after steps 1 and 2, each is unsqueezed from the left side of the shape by necessary number of axes to make both shapes of the same rank.
 
-3. Usual rules of the broadcasting are applied for batch dimensions.
+4. Usual rules of the broadcasting are applied for batch dimensions.
 
-Two attributes, transpose_a and transpose_b specifies embedded transposition for two right-most dimension for the first and the second input tensors correspondingly. It implies swapping of ROW_INDEX_DIM and COL_INDEX_DIM in the corresponding input tensor. Batch dimensions are not affected by these attributes.
+Temporary axes inserted to one-dimensional tensors are removed from the final output shape after multiplying.
+After vector-matrix multiplication, the temporary axis inserted at ROW_INDEX_DIM is removed. After matrix-vector multiplication, the temporary axis inserted at COL_INDEX_DIM is removed.
+Output shape of two equal size vectors multiplication is squeezed to scalar.
+
+Output shape inference logic examples (ND here means bigger than 1D):
+
+* 1D x 1D: `[X] x [X] -> [1, X] x [X, 1] -> [1, 1] => []` (scalar)
+* 1D x ND: `[X] x [B, ..., X, Y] -> [1, X] x [B, ..., X, Y] -> [B, ..., 1, Y] => [B, ..., Y]`
+* ND x 1D: `[B, ..., X, Y] x [Y] -> [B, ..., X, Y] x [Y, 1] -> [B, ..., X, 1] => [B, ..., X]`
+* ND x ND: `[B, ..., X, Y] x [B, ..., Y, Z] => [B, ..., X, Z]`
+
+
+Two attributes, `transpose_a` and `transpose_b` specifies embedded transposition for two right-most dimension for the first and the second input tensors correspondingly. It implies swapping of ROW_INDEX_DIM and COL_INDEX_DIM in the corresponding input tensor. Batch dimensions and 1D tensors are not affected by these attributes.
 
 **Attributes**
 
 * *transpose_a*
 
-  * **Description**: transposes dimensions ROW_INDEX_DIM and COL_INDEX_DIM of the 1st input; 0 means no transpose, 1 means transpose
+  * **Description**: transposes dimensions ROW_INDEX_DIM and COL_INDEX_DIM of the 1st input; **False** means no transpose, **True** means transpose
```

Comment:
Please repeat here that it is ignored if a corresponding input has rank 1.

Reply:
Done